substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -287,8 +311,8 @@ private static ResolvedJavaMethod getOriginal(HostedMethod hostedMethod) {
         if (javaMethod instanceof SubstitutionMethod) {
             SubstitutionMethod substitutionMethod = (SubstitutionMethod) javaMethod;
             javaMethod = substitutionMethod.getAnnotated();
-        } else if (javaMethod instanceof CustomSubstitutionMethod) {
-            javaMethod = ((CustomSubstitutionMethod) javaMethod).getOriginal();
+            // } else if (javaMethod instanceof CustomSubstitutionMethod) {
+            // javaMethod = ((CustomSubstitutionMethod) javaMethod).getOriginal();
```

Comment:
Are these commented out on purpose?

Reply:
Yes, commented-out code should be removed. After all that's what we have a version control system for.

Reply:
Yes, this should be removed.
Also, I have added a comment explaining why this method is expected to traverse to the annotated method for a SubstitutionMethod but otherwise use whatever we unwrap to. I also changed the name to getAnnotatedOrOriginal to reflect what is going on here better.